gobblin-data-management/src/test/java/org/apache/gobblin/data/management/copy/iceberg/IcebergTableMetadataValidatorUtilsTest.java

```diff
@@ -0,0 +1,186 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.gobblin.data.management.copy.iceberg;
+
+import java.io.IOException;
+import java.util.HashMap;
+
+import org.apache.iceberg.PartitionSpec;
+import org.apache.iceberg.TableMetadata;
+import org.apache.iceberg.avro.AvroSchemaUtil;
+import org.apache.iceberg.Schema;
+import org.apache.iceberg.shaded.org.apache.avro.SchemaBuilder;
+import org.testng.Assert;
+import org.testng.annotations.Test;
+
+public class IcebergTableMetadataValidatorUtilsTest {
+  private static final org.apache.iceberg.shaded.org.apache.avro.Schema avroDataSchema1 =
+      SchemaBuilder.record("schema1")
+          .fields()
+          .requiredString("field1")
+          .requiredString("field2")
+          .endRecord();
+  private static final org.apache.iceberg.shaded.org.apache.avro.Schema avroDataSchema2 =
+      SchemaBuilder.record("schema2")
+          .fields()
+          .requiredString("field2")
+          .requiredString("field1")
+          .endRecord();
+  private static final org.apache.iceberg.shaded.org.apache.avro.Schema avroDataSchema3 =
+      SchemaBuilder.record("schema3")
+          .fields()
+          .requiredString("field1")
+          .requiredString("field2")
+          .requiredInt("field3")
+          .endRecord();
+  private static final org.apache.iceberg.shaded.org.apache.avro.Schema avroDataSchema4 =
+      SchemaBuilder.record("schema4")
+          .fields()
+          .requiredInt("field1")
+          .requiredString("field2")
+          .requiredInt("field3")
+          .endRecord();
+  private static final PartitionSpec unpartitionedPartitionSpec = PartitionSpec.unpartitioned();
+  private static final Schema schema1 = AvroSchemaUtil.toIceberg(avroDataSchema1);
+  private static final Schema schema2IsNotSchema1Compat = AvroSchemaUtil.toIceberg(avroDataSchema2);
+  private static final Schema schema3 = AvroSchemaUtil.toIceberg(avroDataSchema3);
+  private static final Schema schema4IsNotSchema3Compat = AvroSchemaUtil.toIceberg(avroDataSchema4);
+  private static final PartitionSpec partitionSpec1 = PartitionSpec.builderFor(schema1)
```

Comment:
actually you did it here -- great!

(if you don't need `avroDataSchemaN` any where else, let's just define the `SchemaBuilder.record`s inline)